include/stdexec/__detail/__shared.hpp

```diff
@@ -329,7 +329,14 @@ namespace stdexec {
         }
 
         STDEXEC_ASSERT(__waiters_copy.front() != __get_tombstone());
-        for (__local_state_base* __item: __waiters_copy) {
+        for (auto __itr = __waiters_copy.begin(); __itr != __waiters_copy.end(); ) {
```

Comment:
I didn't find this fix all that satisfying (it seems a little too "cute"). Open to suggestions here. I considered changing the intrusive slist to handle this better, since the intrusive nature of the data structure leaves the iterator open to such bugs, but that didn't seem like the right thing to do.